subsys/net/l2/openthread/openthread.c

```diff
@@ -353,7 +353,7 @@ void openthread_start(struct openthread_context *ot_context)
 			NET_ERR("Failed to start joiner [%d]", error);
 		}
 
-		return;
+		return error == OT_ERROR_NONE ? 0 : -1;
```

Comment:
it is possible to get better error code than -1?

Reply:
Agreed, -1 might be confusing since typically `errno` value is expected. How about `-EIO`?

Reply:
Fixed